File: Blackjack/Player.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Xml.Linq;

namespace Blackjack2
{
    internal class Player
    {
        public string name { get; private set; }         //player's name
        public int num_of_cards { get; set; }   //number of cards in the player's hand
        public Card[] hand { get; private set; }       //the actual cards in the player's hand
        public int total { get; set; }          //the total value of the cards in the player's hand
        public double money { get; set; }		//the player's amount of money available to bet


        public Player(int starting_amount = 100)
        {
            hand = new Card[5];

            num_of_cards = 0;   //sets initial amount of cards in hand to 0
            money = starting_amount;        //sets intiial amount of money to 100;
            total = 0;			//total value of cards in hand, initially set to 0
        }

        public Player(string n, int starting_amount = 100) : this(starting_amount)
        {
            name = n;
        }

        public void Init()
        {
            Console.Write("Enter your player's name: ");
            name = Console.ReadLine() ?? "Player1";
        }

        public void GetCard(Deck d1)
        {
            hand[num_of_cards] = d1.SendCard();
            num_of_cards++;
        }

        public void DisplayCards()
        {
            int x = 0;
            while (x < num_of_cards)
            {
                hand[x].Draw_Card();
                x++;
            }
        }

    }
}

File: PythonBlackjack/Player.py
class Player:
    def __init__(self, name=None, starting_amount=None):
        if(isinstance(name, str) and len(name) > 0):
            self.__name = name
        else:
            self.__name = input("Enter your name: ") or "Player1"
        
        self.__hand =  [] # Initialize an empty list to hold Card objects
        if(starting_amount != None and isinstance(starting_amount, int) and starting_amount > 0):
            self.__money = starting_amount
        else:
            self.__money = 100  # Invalid format, use default value

    @property
    def name(self):
        return self.__name
    
    @property
    def money(self):
        return self.__money

    @property
    def hand(self):
        return self.__hand

    @property
    def num_of_cards(self):
        return len(self.__hand)
    
    @property
    def total(self):
        total = 0
        for c in self.__hand:
            total += c.get_value(False) # get the total, counting all ACES as 11
        if total > 21:
            for c in self.__hand: #one at a time, switch an ace from an 11 to a 1 until
                if c.rank == 0: #ACE
                    total -= 10
                    if total <= 21:
                        break
        return total

    def get_card(self, card):
        self.__hand.append(card)

    def display_cards(self):
        for card in self.__hand:
            card.draw_card()
    
    def reset_hand(self):
        self.__hand = []
    
    def settle_bet(self, bet):
        self.__money += bet

    def win_amount(self, bet):
        self.__money += bet
    
    def lose_amount(self, bet):
        self.__money -= bet